### ISSUE TITLE
RFC deposit: unification of the typeahead initialization

### ISSUE THREAD

**Issue by kneczaj** [inveniosoftware/invenio#2442]
Hi,

There is typeahead initialization in `form.js` but sometimes there is a need to use something custom with custom data engine. I would like to have a general things, namely: the spinner and auto-saving option for every field, and the initialization of the data engine just for part of them. 

Looking at `AuthorField`:

```python
class AuthorField(WebDepositField, TextField):
    def __init__(self, **kwargs):
        import warnings
        warnings.warn("Field has been deprecated", PendingDeprecationWarning)
        defaults = dict(
            icon='user',
            export_key='authors[0].full_name',
            autocomplete=orcid_authors,
            widget_classes="form-control"
        )
        defaults.update(kwargs)
        super(AuthorField, self).__init__(**defaults)
```

There is `autocomplete` parameter with a function which returns suggestions. I would like to add a parameter to just add the mentioned general stuff and leave the data engine initialization for custom javascript. In version 0.11 and later the spinner can be rewritten to react on events of typeahead https://github.com/twitter/typeahead.js/issues/166 which make it quite easy to do.

At html:
Right now in html `'data-autocomplete' = "1"` means that the standard autocomplete from `form.js` will be used. At html side i propose to change it to `'data-autocomplete' = "default"` to get default autocomplete initialized, and `data-autocomplete` equal whatever else to have just the events connected. A field with empty `data-autocomplete` would not be affected by typeahead at all.

At python side: 
Right now `autocomplete` parameter is a function so maybe a function would be mapped to proposed `'data-autocomplete' = "default"` and a string to `'data-autocomplete' = "the string value"`

Tell me what you think about it.

**Comment by tiborsimko** [comment]
... and in PR #2521. @jmartinm please add `closes` directive as appropriate.

**Comment by jirikuncar** [comment]
Closed via #2521.
